### ISSUE TITLE
Build API Documentations

### ISSUE THREAD

**Issue by gengliangwang** [databrickslabs/pyspark-ai#83]
Currently, there are no API documents. We should start creating one

**Comment by nayan2167** [comment]
Hi @gengliangwang I would like to work on this, can you explain it a bit more?

**Comment by gengliangwang** [comment]
@nayan2167 Yes it will be a simple Python doc website based on markdown files and our example notebooks.

**Comment by Pranavchiku** [comment]
I am planning to go ahead with `mkdocs` if it works.

**Comment by gengliangwang** [comment]
@Pranavchiku let's go with sphinx

**Comment by Pranavchiku** [comment]
Perfect, I will make a draft PR soon.

**Comment by SemyonSinchenko** [comment]
If help is needed here I can do it. Recently I added generation of API docs for [Quinn](https://github.com/MrPowers/quinn) PySpark helpers library, so I know how to do it.

It requires the following:
1. Add some dependencies to `pyproject.toml`: `mkdocstrings-python`, `mkdocs-gen-files`, `mkdocs-literate-nav`
2. Add a simple python script like [this one](https://github.com/MrPowers/quinn/blob/main/docs/gen_ref_pages.py) to `docs`
3. Add a link to this script to `mcdocs.yml` like it was done [here](https://github.com/MrPowers/quinn/blob/main/mkdocs.yml#L10)

In runtime it will generate the documentation for all Public API and `SUMMARY.md` file with right links structure. You can see how it looks like [here](https://mrpowers.github.io/quinn/reference/SUMMARY/).

**Comment by SemyonSinchenko** [comment]
May I help somehow here? @Pranavchiku if you do not have time to do it now, just notify me please.

**Comment by SemyonSinchenko** [comment]
@gengliangwang Do you know if anyone is still working on this? Can I help here?

**Comment by gengliangwang** [comment]
@SemyonSinchenko Please help on this one. Thank you in advance!
